src/client/Client.cc

```diff
@@ -1461,13 +1460,13 @@ mds_rank_t Client::choose_target_mds(MetaRequest *req, Inode** phash_diri)
       }
     }
   
-    if (req->auth_is_best())
-      cap = in->auth_cap;
-    if (!cap && !in->caps.empty())
-      cap = in->caps.begin()->second;
-    if (!cap)
+    if (req->auth_is_best()) {
```

Comment:
should be 
if (in->auth_cap && req->auth_is_best())

Reply:
Great catch, thanks Zheng.